Port Radius Profile Configuration

rclee opened the pull request:
- Creates new class MMEEventsService to manage the creation and distribution of MMEEventsConfiguration objects/configurations.
- Expands MMEEventsConfiguration to include a variable hibernation radius configuration and potentially additional configurations in the future. A user may access the variable radius by setting both "MMEEventsProfile" (with the appropriate key "VariableGeofence") and "MMEGeofenceRadius" with a desired radius (within a range we've set) in the applications plist. If "MMEEventsProfile" is set in plist without choosing a radius then a median radius will be the default. 
- Modifies MMELocationManager and MMEEventsManager to use MMEEventsService configuration
- Adds and repairs cedar tests

rclee committed:
Port Radius Profile Configuration

- Creates new class MMEEventsService to manage the creation and distribution of MMEEventsConfiguration objects/configurations.
- Expands MMEEventsConfiguration to include a variable hibernation radius configuration and potentially additional configurations in the future. A user may access the variable radius by setting both "MMEEventsProfile" (with the appropriate key "VariableGeofence") and "MMEGeofenceRadius" with a desired radius (within a range we've set) in the applications plist. If "MMEEventsProfile" is set in plist without choosing a radius then a median radius will be the default. 
- Modifies MMELocationManager and MMEEventsManager to use MMEEventsService configuration
- Adds and repairs cedar tests

rclee committed:
Port Radius Profile Configuration

- Creates new class MMEEventsService to manage the creation and distribution of MMEEventsConfiguration objects/configurations.
- Expands MMEEventsConfiguration to include a variable hibernation radius configuration and potentially additional configurations in the future. A user may access the variable radius by setting both "MMEEventsProfile" (with the appropriate key "VariableGeofence") and "MMEGeofenceRadius" with a desired radius (within a range we've set) in the applications plist. If "MMEEventsProfile" is set in plist without choosing a radius then a median radius will be the default. 
- Modifies MMELocationManager and MMEEventsManager to use MMEEventsService configuration
- Adds and repairs cedar tests

boundsj reviewed (approved):
This looks good to me. Let's hold off on merging for now until we have confirmation of the need for this change and a spec for the upper bound (kHibernationRadiusWide). Thanks!

rclee committed:
Merge pull request #39 from mapbox/develop

Visit Event Collection

rclee committed:
Update version to 0.4.0

friedbunny committed:
Switch to @available for version checking

This requires Xcode 9.x.

rclee committed:
Fix bug with geofence creation related to custom profile

geofence were being created too often when the profile was set to a wide geofence radius.

rclee committed:
Fix bug with geofence creation related to custom profile

geofence were being created too often when the profile was set to a wide geofence radius.

rclee committed:
fixes merge conflict
